api/v1beta2/foundationdbcluster_types.go

```diff
@@ -1677,24 +1677,27 @@ type RoutingConfig struct {
 	DNSDomain *string `json:"dnsDomain,omitempty"`
 }
 
-// RequiredAddressSet provides settings for which addresses we need to listen
+// Requiredaddressset provides settings for which addresses we need to listen
 // on.
 type RequiredAddressSet struct {
 	// TLS defines whether we need to listen on a TLS address.
 	TLS bool `json:"tls,omitempty"`
 
 	// NonTLS defines whether we need to listen on a non-TLS address.
-	NonTLS bool `json:"nonTLS,omitempty"`
+	NonTLS bool `jsogn:"nonTLS,omitempty"`
 }
 
+// ProcessGroup -> ContainerName
+type CrashLoopList map[string](map[string]None)
+
 // BuggifyConfig provides options for injecting faults into a cluster for testing.
 type BuggifyConfig struct {
 	// NoSchedule defines a list of process group IDs that should fail to schedule.
 	NoSchedule []string `json:"noSchedule,omitempty"`
 
 	// CrashLoops defines a list of process group IDs that should be put into a
 	// crash looping state.
-	CrashLoop []string `json:"crashLoop,omitempty"`
+	CrashLoop CrashLoopList `json:"crashLoop,omitempty"`
```

Comment:
We have to add a new CrashLoop list for adding the new feature otherwise we break our existing API. Internally we can convert the old format into the new format and define that only one of the two list are allowed to be set.